backend/tests/unit/dal/Services/PropertyServiceTest.cs

```diff
@@ -0,0 +1,306 @@
+            {
+                new object[] { new ParcelFilter(50, 25, 50, 20), 1 },
+                new object[] { new ParcelFilter(50, 24, 50, 26), 0 },
+                new object[] { new ParcelFilter() { Agencies = new int[] { 3 } }, 1 },
+                new object[] { new ParcelFilter() { ClassificationId = 2 }, 1 },
+                new object[] { new ParcelFilter() { Description = "Description" }, 1 },
+                new object[] { new ParcelFilter() { Municipality = "Municipality" }, 1 },
+                new object[] { new ParcelFilter() { Zoning = "Zoning" }, 1 },
+                new object[] { new ParcelFilter() { ZoningPotential = "ZoningPotential" }, 1 }
+            };
+
+        public static IEnumerable<object[]> BuildingFilters =>
+            new List<object[]>
+            {
+                new object[] { new BuildingFilter(50, 25, 50, 20), 1 },
+                new object[] { new BuildingFilter(50, 24, 50, 26), 0 },
+                new object[] { new BuildingFilter() { Agencies = new int[] { 3 } }, 1 },
+                new object[] { new BuildingFilter() { ClassificationId = 2 }, 1 },
+                new object[] { new BuildingFilter() { Description = "Description" }, 1 },
+                new object[] { new BuildingFilter() { Tenancy = "BuildingTenancy" }, 1 },
+                new object[] { new BuildingFilter() { ConstructionTypeId = 2 }, 1 },
+                new object[] { new BuildingFilter() { PredominateUseId = 2 }, 1 },
+                new object[] { new BuildingFilter() { MinRentableArea = 100 }, 1 },
+                new object[] { new BuildingFilter() { MinRentableArea = 50, MaxRentableArea = 50 }, 1 }
+            };
+
+        public static IEnumerable<object[]> AllPropertyFilters =>
+            new List<object[]>
+            {
+                new object[] { new AllPropertyFilter(50, 25, 50, 20), 2 },
+                new object[] { new AllPropertyFilter(50, 24, 50, 26), 0 },
+                new object[] { new AllPropertyFilter() { Agencies = new int[] { 3 } }, 7 },
+                new object[] { new AllPropertyFilter() { ClassificationId = 2 }, 2 },
+                new object[] { new AllPropertyFilter() { Description = "Description" }, 10 },
+                new object[] { new AllPropertyFilter() { Municipality = "Municipality" }, 1 },
+                new object[] { new AllPropertyFilter() { Tenancy = "BuildingTenancy" }, 1 },
+                new object[] { new AllPropertyFilter() { Zoning = "Zoning" }, 10 },
+                new object[] { new AllPropertyFilter() { ZoningPotential = "ZoningPotential" }, 10 },
+                new object[] { new AllPropertyFilter() { ConstructionTypeId = 2 }, 1 },
+                new object[] { new AllPropertyFilter() { PredominateUseId = 2 }, 1 },
+                new object[] { new AllPropertyFilter() { MinRentableArea = 100 }, 1 },
+                new object[] { new AllPropertyFilter() { MinRentableArea = 50, MaxRentableArea = 50 }, 1 },
+            };
```

Comment:
I love all the unit tests covering the filtering functionality. 
Given that this PR is mostly about paginated requests - would you consider adding a couple tests that cover pagination? 

Something like -  "there's 10 properties, if I set `filter.quantity = 5` and `filter.page = 2`, I should see the last five?"

Reply:
That is correct